### PR TITLE
fix(kubectl-plugin): non-zero exit code on error

### PR DESCRIPTION
**Pull request by datacore-vvarakantham**
Returning non zero exit code with an error from plugin with kubectl command.

### CODE REVIEW

**Review by Abhinandan-Purkait — APPROVED** [review]
Should it be rather a wrapper function which logs an error message and returns an exit code, rather than calling exit everywhere. In that way we can reuse the same method? @tiagolobocastro wdyt?

**Comment by tiagolobocastro** [comment]
> Should it be rather a wrapper function which logs an error message and returns an exit code, rather than calling exit everywhere. In that way we can reuse the same method? @tiagolobocastro wdyt?

hmm yeah...
I think we can take one step further, and simply return an error up the call stack

File: control-plane/plugin/src/operations.rs [hunk 1]
```diff
@@ -1,6 +1,11 @@
-use crate::resources::{utils, CordonResources, DrainResources, GetResources, ScaleResources};
+use crate::resources::{
+    error::Error, utils, CordonResources, DrainResources, GetResources, ScaleResources,
+};
 use async_trait::async_trait;
 
+/// Result wrapper for kubectl commands.
+pub type KubectlResult = Result<(), Error>;
```
Comment (tiagolobocastro) [inline control-plane/plugin/src/operations.rs]: there's no Kubectl here
```suggestion
/// Result wrapper for plugin commands.
pub type PluginResult = Result<(), Error>;
```

File: control-plane/plugin/src/bin/rest-plugin/main.rs [hunk 2]
```diff
@@ -133,4 +133,9 @@ async fn execute(cli_args: CliArgs) {
             }
         },
     };
+
+    let _ = result.map_err(|error| {
+        eprintln!("{error}");
```
Comment (tiagolobocastro) [inline control-plane/plugin/src/bin/rest-plugin/main.rs:138]: if output json/yaml then please change print accordingly

File: control-plane/plugin/src/bin/rest-plugin/main.rs [hunk 3]
```diff
@@ -133,4 +133,9 @@ async fn execute(cli_args: CliArgs) {
             }
         },
     };
+
+    let _ = result.map_err(|error| {
```
Comment (tiagolobocastro) [inline control-plane/plugin/src/bin/rest-plugin/main.rs]:
```suggestion
    if let Err(error) = result {
```

**Review by tiagolobocastro — APPROVED** [review]

**Comment by datacore-vvarakantham** [comment]
bors merge

**Comment by datacore-vvarakantham** [comment]
bors merge

**Pull request merged**